### ISSUE TITLE
BUG : FILTER fails when using negation (!) on variable whose name start with "in"

### ISSUE THREAD

**Issue by Cajuteq** [arangodb/arangodb#17367]
## My Environment

* __ArangoDB Version__:  3.10.0      
* __Deployment Mode__: Single Server      
* __Deployment Strategy__: ArangoDB Official Docker with Docker-compose  
* __Configuration__: [Low memory recommandations from the doc](https://www.arangodb.com/docs/stable/tutorials-reduce-memory-footprint.html)           
* __Infrastructure__: OVH VPS              
* __Operating System__: Debian 10       
* __Total RAM in your machine__: 8Gb      
* __Disks in use__: HDD(?)       
* __Used Package__: Docker - official             

## Component, Query & Data

__Affected feature__: AQL query

__AQL query (if applicable)__:
```
FOR id IN ["Users/abc123"]

    FOR instance, e IN 1..1 OUTBOUND id IS_IN_INS
      FILTER !instance.HIDE
        RETURN instance
```
```
FOR id IN ["Users/abc123"]

    FOR instance, e IN 1..1 OUTBOUND id IS_IN_INS
      FILTER instance.HIDE
        RETURN instance
```
```
FOR id IN ["Users/abc123"]

    FOR istance, e IN 1..1 OUTBOUND id IS_IN_INS
      FILTER !istance.HIDE
        RETURN istance
```
__AQL explain and/or profile (if applicable)__:

```
Query String (147 chars, cacheable: true):
 FOR id IN ["Users/abc123"]
     FOR v, info IN 1..1 OUTBOUND id IS_IN_INS
       FILTER info.HIDE
         RETURN info
 

Execution plan:
 Id   NodeType            Est.   Comment
  1   SingletonNode          1   * ROOT
  2   CalculationNode        1     - LET #3 = [ "Users/abc123" ]   /* json expression */   /* const assignment */
  3   EnumerateListNode      1     - FOR id IN #3   /* list iteration */
  4   TraversalNode          5       - FOR /* vertex optimized away */, info  /* edge */ IN 1..1  /* min..maxPathDepth */ OUTBOUND id /* startnode */  IS_IN_INS
  5   CalculationNode        5         - LET #7 = info.`HIDE`   /* attribute expression */
  6   FilterNode             5         - FILTER #7
  7   ReturnNode             5         - RETURN info

Indexes used:
 By   Name   Type   Collection   Unique   Sparse   Cache   Selectivity   Fields        Stored values   Ranges
  4   edge   edge   IS_IN_INS    false    false    false       17.22 %   [ `_from` ]   [  ]            base OUTBOUND

Traversals on graphs:
 Id  Depth  Vertex collections  Edge collections  Options                                  Filter / Prune Conditions
 4   1..1                       IS_IN_INS         uniqueVertices: none, uniqueEdges: path  FILTER info.`HIDE`       

Optimization rules applied:
 Id   RuleName
  1   optimize-traversals

41 rule(s) executed, 1 plan(s) created
```

__Dataset__:

A Users collection linked to a Instances with a IS_IN_INS edge

Each User can be linked to multiple Instances and an Instance usually contains several Users

A User has the ability to HIDE an Instance he participates in and some UI will allow him to list the Instances he has while HIDE-ing the one he wants.

## Steps to reproduce

1. create the Users, Instances and IS_IN_INS collections
2. populate the collection with dummy datas linking some Users to some Instances
3. try to execute the first query

__Problem__:
The first query give 
  `Query: AQL: syntax error, unexpected not in operator near '!instance.HIDE RETURN in...' at position 4:14 (while parsing)`

Both the second and third queries work fine (although the second one obviously give back inverted results)

__Expected result__:
We should be able to use FILTER with NOT (!) operand even when our variables names starts with "in" or the doc should clearly states that it is prohibited.

__Note__:
The case was raised following an upgrade from 3.6.1 to 3.10.0 and I am sure the exact same query was fully functional in 3.6.1 so I suggest looking for optimization made in this time-frame.

**Comment by Supytalp** [comment]
Adding parentheses in order to break the "!in" character chain solves the issue.

It looks like AQL doesn't recognize the "in" as being part of the object's name, and directly converts "!in*" as "not in *".

**Comment by Simran-B** [comment]
I can confirm that any `in` after `!` or `NOT` is interpreted as `NOT IN` operator, even if there is no word boundary after `in`.
This is a general problem with the lexer, not restricted to graph traversals.

Whitespace between `NOT` and `in...` does not help, e.g. `LET int = 0 RETURN ! int`
As a workaround, you can either use parentheses like @Supytalp suggests, or wrap the variable name in forward or backticks, e.g. `LET int = 0 RETURN ! ´int´`

**Comment by Simran-B** [comment]
A fix will be released with v3.10.2 in the coming weeks.
